provider/marathon/marathon.go

```diff
@@ -274,8 +357,24 @@ func (p *Provider) getLabel(application marathon.Application, label string) (str
 	return "", false
 }
 
-func (p *Provider) getPort(task marathon.Task, application marathon.Application) string {
-	port, err := processPorts(application, task)
+// Extract port from labels for a given service and a given marathon application
+func (p *Provider) getPort(task marathon.Task, application marathon.Application, serviceName string) string {
+	if value, ok := getServiceProperty(application, serviceName, "port"); ok {
```

Comment:
I'll take a stab at it...  Now, that `getLabel` always falls back to do application generic label lookup if the service label not existant, that might help collapse some of the logic, and I'll only ever need to handle 2 generic cases... port and portIndex, without having to worry about services...